src/contracts/libraries/SlashingAccountingUtils.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+/**
+ * @title Library of utilities for calculations related to slashing in EigenLayer
+ * @author Layr Labs, Inc.
+ * @notice Terms of Service: https://docs.eigenlayer.xyz/overview/terms-of-service
+ */
+library SlashingAccountingUtils {
+
+    // TODO: set this based on payments epochs
+    uint256 internal constant EPOCH_GENESIS_TIMESTAMP = 1606824000;
+
+    // TODO: define appropriately
+    uint256 internal constant EPOCH_LENGTH_SECONDS = 2 weeks;
+
+    uint256 internal constant SHARE_CONVERSION_SCALE = 1e18;
+
+    // an amount of shares over this will cause overflow when multiplying by `SHARE_CONVERSION_SCALE`
+    uint256 internal constant MAX_VALID_SHARES = type(uint96).max;
+
+    uint256 internal constant BIPS_FACTOR = 10000;
+
+    // TODO: explain this better. basically seems like we may need to set some max factor beyond which shares are just zeroed out
+    uint256 internal constant MAX_SCALING_FACTOR = type(uint256).max / (MAX_VALID_SHARES * SHARE_CONVERSION_SCALE);
+
+    function getEpochFromTimestamp(uint256 timestamp) internal pure returns (int256) {
```

Comment:
Does it make sense to have negative epochs if a timestamp before GENESIS is specified? I guess it's cleaner than having the view function revert

Reply:
that was my line of reasoning as well.

Reply:
eh, i prefer revert. signed integers scare me. maybe skill issue tho